File: src/kyu_7/simple_string_reversal/csharp/simple_string_reversal.cs
public class Solution
{
    public static string solve(string s)
    {
        var a = s.ToCharArray();

        for (int i = 0, j = s.Length-1; j >= 0; i++, j--)
        {
          if (s[i] == ' ') i++;
          if (s[j] == ' ') j--;
          a[i] = s[j];
        }

        return new string(a);
    }
}

File: src/kyu_7/simple_string_reversal/cpp/simple_string_reversal.cpp
using namespace std;

string solve(string s) {
  string reversed;
  for (int i = s.length() - 1; i >= 0; i--)
  {
    if (s[i] != ' ')
    {
      reversed += s[i];
    }
    if (s[reversed.length()] == ' ')
    {
      reversed += ' ';
    }
  }
  return reversed;
}